internal/command/views/json/diagnostic.go

```diff
@@ -38,6 +38,7 @@ type Diagnostic struct {
 	Address  string             `json:"address,omitempty"`
 	Range    *DiagnosticRange   `json:"range,omitempty"`
 	Snippet  *DiagnosticSnippet `json:"snippet,omitempty"`
+	Extra    interface{}        `json:"extra,omitempty"`
```

Comment:
A change to this externally-facing json format will require an update to its docs page (https://developer.hashicorp.com/terraform/cli/commands/validate#json), and possibly a bump to its minor version number. 

(But, I asked around about how seriously we take the minor version guarantees in these json formats, and the answer was inconclusive. We do at least want it documented, though.)

Reply:
Included 👍  https://github.com/hashicorp/terraform/pull/34943/commits/ed4a71c04ff7dea8cd150924834bd235ff5e7542

I was brief with the explanation since its relatively broad, happy to add more detail if needed 😁